storage/localstore/localstore.go

```diff
@@ -133,6 +135,10 @@ type Options struct {
 	// MetricsPrefix defines a prefix for metrics names.
 	MetricsPrefix string
 	Tags          *chunk.Tags
+	// PutSetCheckFunc is a function called after a Put of a chunk
+	// to verify whether that chunk needs to be Set and added to
+	// garbage collection index too
+	PutSetCheckFunc func([]byte) bool
```

Comment:
This function has to do with adding a chunk to the GC, not to any Set method call. A name like `PutToGCCheck` may be more appropriate, or something similar?